src/pages/individual/EditIndividualMetadata.jsx

```diff
@@ -123,12 +146,78 @@ export default function EditIndividualMetadata({
           loading={loading}
           display="primary"
           onClick={async () => {
-            const currentPatchableValues = pick(defaultFieldValues, [
-              'names',
-            ]);
+            // validation
+            const requiredFieldErrors = metadata.reduce(
+              (memo, schema) => {
+                if (!schema?.required) return memo;
+
+                const formObject = schema.customField
+                  ? customFieldValues
+                  : defaultFieldValues;
+
+                if (!formObject[schema.name]) {
+                  const intlFieldName = intl.formatMessage({
+                    id: schema.labelId,
+                  });
+                  memo.push(
+                    intl.formatMessage(
+                      { id: 'IS_REQUIRED' },
+                      { field: intlFieldName },
+                    ),
+                  );
+                }
+
+                return memo;
+              },
+              [],
+            );
+
+            setFormErrors(requiredFieldErrors);
+            if (requiredFieldErrors.length > 0) return;
+
+            const names = [];
+            // Always include the defaultName. If it does not already exist, add it.
+            // Otherwise, replace it.
+            const defaultNameGuid = deriveNameGuid(
+              metadata,
+              'defaultName',
+            );
+            const defaultNameProperty = defaultNameGuid
+              ? {
+                  op: 'replace',
+                  value: defaultFieldValues.defaultName,
+                  guid: defaultNameGuid,
+                }
+              : {
+                  op: 'add',
+                  value: defaultFieldValues.defaultName,
+                  context: 'defaultName',
+                };
+            names.push(defaultNameProperty);
```

Comment:
might be a little cleaner to declare `names` at this point instead
```
const names = [defaultNameProperty] 
```

Reply:
Fixed in https://github.com/WildMeOrg/codex-frontend/pull/287/commits/3c4e9c3a4c2dd44627eec3fe3a36946dd7e8c0ed